martian/src/lib.rs

```diff
@@ -300,8 +305,31 @@ pub fn make_mro_string(mro_registry: &[StageMro]) -> String {
     }
     mro_string.pop();
 
-    format!(
-        "{}{}{}{}",
-        MRO_HEADER, filetype_header, struct_header, mro_string
-    )
+    if header_comment.is_empty() {
+        format!(
+            "#
+# Code generated by {}.  DO NOT EDIT.
+#
+
+{}{}{}",
+            get_generator_name(),
+            filetype_header,
+            struct_header,
+            mro_string
+        )
+    } else {
+        format!(
+            "{}
```

Comment:
It might be good to check that all the lines in `header_comment` start with `#` here?

Reply:
Honestly not terribly interested in figuring out how to do that in rust; as far as I'm concerned if a user wants to generate invalid MRO they're welcome to do so.  Open to suggestions, though.

Reply:
Done.  Please check if I did it properly.

Reply:
Awesome! Looks good to me.